web/src/components/Header.tsx

```diff
@@ -107,13 +109,15 @@ const Header = () => {
           </NavLink>
           {!isVisitorMode && (
             <>
-              <button
-                id="header-ask-ai"
-                className="px-4 pr-5 py-2 rounded-lg flex flex-row items-center text-lg text-gray-800 dark:text-gray-300 hover:bg-white hover:shadow dark:hover:bg-zinc-700"
-                onClick={() => showAskAIDialog()}
-              >
-                <Icon.Bot className="mr-3 w-6 h-auto opacity-70" /> {t("ask-ai.title")}
-              </button>
+              {showAskAI && (
```

Comment:
How about hide it when openai config is undefined or openai api key config is undefined? So we don't need to introduce a new setting item.

Reply:
I put the openai config in with the rest of the global state and I'm checking it's not empty in showAskAI